doc/nrf/config_and_build/bootloaders_and_dfu/fw_update.rst

```diff
@@ -331,16 +340,56 @@ If this image has, in order of precedence, a *major*, *minor*, or *revision* val
    The optional label or build number specified after the ``+`` character is ignored when evaluating the version.
    An existing application image with version ``0.1.2+3`` can be overwritten by an uploaded image with ``0.1.2+2``, but not by one with ``0.1.1+2``.
 
-Hardware-based downgrade prevention
------------------------------------
+.. _ug_fw_update_downgrade_protection_hw:
+.. _bootloader_monotonic_counter:
+
+Hardware-based downgrade protection
+===================================
 
-Hardware-based downgrade prevention is implemented with a counter value stored in the *provision* partition of the non-volatile storage.
+.. bootloader_monotonic_counter_start
 
-See the :ref:`bootloader_monotonic_counter` section for how to enable and configure this feature.
+You can implement hardware-based downgrade protection using a non-volatile monotonic counter.
+
+Counter updates are written to slots in the *Provision* area, with each new counter update occupying a new slot.
+For this reason, the number of counter updates, and therefore firmware version updates, is limited.
+
+Using a counter is optional and can be configured for the application using configuration options.
+You can also configure the supported number of updates, but the number is limited by the size of the *Provision* area and how much of that area is taken up by other features, like public key hashes.
+In addition, you can configure what firmware version of the image you want to boot.
+
+.. bootloader_monotonic_counter_end
+
+Special handling is needed when updating the S1 variant of an image.
+See :ref:`ug_bootloader_adding_presigned_variants` for details.
+
+To set options for child images, such as NSIB and MCUboot, see the :ref:`ug_multi_image_variables` section.
+
+Using |NSIB|
+============
+
+.. bootloader_monotonic_counter_nsib_start
+
+For |NSIB|, the configuration options for implementing the anti-rollback protection with monotonic counter are the following:
+
+* :kconfig:option:`CONFIG_SB_MONOTONIC_COUNTER`
+* :kconfig:option:`CONFIG_SB_NUM_VER_COUNTER_SLOTS`
+
+.. bootloader_monotonic_counter_nsib_end
+
+.. _ug_fw_update_image_versions_mcuboot:
+
+Using MCUboot
+=============
+
+For MCUboot, the configuration options for implementing the anti-rollback protection with monotonic counter are the following:
+
+* :kconfig:option:`CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION`
```

Comment:
@sigvartmh , would you know?

Reply:
I've not touched this configuration, so I'm unsure. I would preface it with that it probably does not work with (mcuboot image slot > 2) network core upgrades, as the last time someone tried this, it did not work.

Reply:
So that the rolling counter is updated next to the version and both need to be maintained by the user?

Reply:
The way it work(I think) I've not looked at this part of MCUBoot in a while is that once you update with a firmware version which is greater than your current version and is proven working the rolling counter is incremented. This means you will not be able to rollback to the previous version.

I also think this only works with XiP and it rotates between the slots primary/secondary depending on the rolling counter. So the user doesn't need to maintain the rolling counter only semantic versioning of MCUBoot.

Reply:
Thanks for clarification. Could you add these few sentences to documentation, please? (we could do it in a separate PR)

Reply:
FYI: @MarekPieta @greg-fer I see that it works for swaps too. The user doesn't need to maintain the security counter themselves if you don't provide a value as it would be calculated based on the version number see:

https://github.com/nrfconnect/sdk-mcuboot/blob/adab597a0eb0eb9c030a7b797748a49ca89988c2/scripts/imgtool/image.py#L185

If this option is set CONFIG_MCUBOOT_HW_DOWNGRADE_PREVENTION_COUNTER_VALUE 
this value is provided:  set(imgtool_security_counter --security-counter ${CONFIG_MCUBOOT_HW_DOWNGRADE_PREVENTION_COUNTER_VALUE})

Maybe just keep it as auto and recommend the users to have versioning 0.0.1 on their first software and only increase revision. Sounds a bit weird as you wouldn't keep a semantic versioning scheme with this.

https://github.com/nrfconnect/sdk-mcuboot/blob/adab597a0eb0eb9c030a7b797748a49ca89988c2/scripts/imgtool/main.py#L256C4-L256C4

Related MCUBoot code:
https://github.com/nrfconnect/sdk-mcuboot/blob/adab597a0eb0eb9c030a7b797748a49ca89988c2/boot/bootutil/src/loader.c#L923

So my bad seems that the counter value is being used just set by the script automatically.

Reply:
Does this mean we have to change the CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION option to be enabled by default ("maybe just keep it as auto")?

Also, if we need to update the information about image version, we need to do this in the "Configuring image version with MCUboot" section.

@MarekPieta , @sigvartmh ?

Reply:
> Does this mean we have to change the CONFIG_MCUBOOT_HARDWARE_DOWNGRADE_PREVENTION option to be enabled by default ("maybe just keep it as auto")?
> 
> Also, if we need to update the information about image version, we need to do this in the "Configuring image version with MCUboot" section.
> 
> @MarekPieta , @sigvartmh ?

Enabling MCUboot downgrade prevention by default would change current default behavior and may be a bit confusing for end users (if we opt for it, we would need to clearly inform about the change in a visible place I think).

Updating doc may be a good idea (either in this or subsequent PR).